File: Api_java/estudianteApi/src/main/java/com/example/estudianteApi/Entidad/Material.java
package com.example.estudianteApi.Entidad;

import jakarta.persistence.*;
import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "materiales")
public class Material {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_material")
    private Integer idMaterial;

    @Column(name = "nombre", nullable = false, length = 200)
    private String nombre;

    @Column(name = "descripcion", columnDefinition = "TEXT")
    private String descripcion;

    @Column(name = "categoria", length = 50)
    private String categoria;

    @Column(name = "precio")
    private BigDecimal precio = new BigDecimal("0.00");

    @Column(name = "stock")
    private Integer stock = 0;

    @Column(name = "imagen_url", length = 500)
    private String imagenUrl;

    @Column(name = "autor", length = 200)
    private String autor;

    @Column(name = "editorial", length = 200)
    private String editorial;

    @Column(name = "disponible")
    private Boolean disponible = true;

    @Column(name = "fecha_creacion")
    private LocalDateTime fechaCreacion = LocalDateTime.now();

    // Constructores
    public Material() {
    }

    public Material(String nombre, String descripcion, String categoria, BigDecimal precio,
            Integer stock, String autor, String editorial) {
        this.nombre = nombre;
        this.descripcion = descripcion;
        this.categoria = categoria;
        this.precio = precio;
        this.stock = stock;
        this.autor = autor;
        this.editorial = editorial;
    }

    // Getters y Setters
    public Integer getIdMaterial() {
        return idMaterial;
    }

    public void setIdMaterial(Integer idMaterial) {
        this.idMaterial = idMaterial;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public String getDescripcion() {
        return descripcion;
    }

    public void setDescripcion(String descripcion) {
        this.descripcion = descripcion;
    }

    public String getCategoria() {
        return categoria;
    }

    public void setCategoria(String categoria) {
        this.categoria = categoria;
    }

    public BigDecimal getPrecio() {
        return precio;
    }

    public void setPrecio(BigDecimal precio) {
        this.precio = precio;
    }

    public Integer getStock() {
        return stock;
    }

    public void setStock(Integer stock) {
        this.stock = stock;
    }

    public String getImagenUrl() {
        return imagenUrl;
    }

    public void setImagenUrl(String imagenUrl) {
        this.imagenUrl = imagenUrl;
    }

    public String getAutor() {
        return autor;
    }

    public void setAutor(String autor) {
        this.autor = autor;
    }

    public String getEditorial() {
        return editorial;
    }

    public void setEditorial(String editorial) {
        this.editorial = editorial;
    }

    public Boolean getDisponible() {
        return disponible;
    }

    public void setDisponible(Boolean disponible) {
        this.disponible = disponible;
    }

    public LocalDateTime getFechaCreacion() {
        return fechaCreacion;
    }

    public void setFechaCreacion(LocalDateTime fechaCreacion) {
        this.fechaCreacion = fechaCreacion;
    }
}

File: flutter_application_1/lib/src/models/material.dart
class Material {
  final int idMaterial;
  final String nombre;
  final String? descripcion;
  final String? categoria;
  final double? precio;
  final int? stock;
  final String? imagenUrl;
  final String? autor;
  final String? editorial;
  final bool? disponible;
  final String? fechaCreacion;

  Material({
    required this.idMaterial,
    required this.nombre,
    this.descripcion,
    this.categoria,
    this.precio,
    this.stock,
    this.imagenUrl,
    this.autor,
    this.editorial,
    this.disponible,
    this.fechaCreacion,
  });

  factory Material.fromJson(Map<String, dynamic> json) {
    return Material(
      idMaterial: json['idMaterial'] ?? 0,
      nombre: json['nombre'] ?? '',
      descripcion: json['descripcion'],
      categoria: json['categoria'],
      precio: json['precio']?.toDouble(),
      stock: json['stock'],
      imagenUrl: json['imagenUrl'],
      autor: json['autor'],
      editorial: json['editorial'],
      disponible: json['disponible'],
      fechaCreacion: json['fechaCreacion'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'idMaterial': idMaterial,
      'nombre': nombre,
      'descripcion': descripcion,
      'categoria': categoria,
      'precio': precio,
      'stock': stock,
      'imagenUrl': imagenUrl,
      'autor': autor,
      'editorial': editorial,
      'disponible': disponible,
      'fechaCreacion': fechaCreacion,
    };
  }

  // Helper para verificar disponibilidad
  bool get estaDisponible => (disponible ?? false) && (stock ?? 0) > 0;

  // Helper para obtener precio formateado
  String get precioFormateado => '\$${precio?.toStringAsFixed(2) ?? '0.00'}';
}
